pkg/steps/new_e2e.go

```diff
@@ -0,0 +1,147 @@
+)
+
+type newE2ETestStep struct {
+	name             string
+	podClient        PodClient
+	jobSpec          *api.JobSpec
+	pre, tests, post []api.TestStep
+}
+
+func NewE2ETestStep(
+	config api.TestStepConfiguration,
+	podClient PodClient,
+	jobSpec *api.JobSpec,
+) (api.Step, error) {
+	return newNewE2ETestStep(config, podClient, jobSpec), nil
+}
+
+func newNewE2ETestStep(
+	config api.TestStepConfiguration,
+	podClient PodClient,
+	jobSpec *api.JobSpec,
+) *newE2ETestStep {
+	return &newE2ETestStep{
+		name:      config.As,
+		podClient: podClient,
+		jobSpec:   jobSpec,
+		pre:       config.PreSteps,
+		tests:     config.Steps,
+		post:      config.PostSteps,
+	}
+}
+
+func (s *newE2ETestStep) Inputs(ctx context.Context, dry bool) (api.InputDefinition, error) {
+	return nil, nil
+}
+
+func (s *newE2ETestStep) Run(ctx context.Context, dry bool) error {
+	if dry {
+		return nil
+	}
+	var errs []error
+	if err := s.runSteps(ctx, s.pre); err != nil {
+		errs = append(errs, fmt.Errorf("%q pre steps failed: %v", s.name, err))
+	} else {
+		if err := s.runSteps(ctx, s.tests); err != nil {
+			errs = append(errs, fmt.Errorf("%q test steps failed: %v", s.name, err))
+		}
+	}
+	if err := s.runSteps(ctx, s.post); err != nil {
+		errs = append(errs, fmt.Errorf("%q post steps failed: %v", s.name, err))
+	}
+	return utilerrors.NewAggregate(errs)
+}
+
+func (s *newE2ETestStep) Done() (bool, error) { return false, nil }
+func (s *newE2ETestStep) Name() string        { return s.name }
+func (s *newE2ETestStep) Description() string {
+	return fmt.Sprintf("Run cluster install %s", s.name)
+}
+
+func (s *newE2ETestStep) Requires() (ret []api.StepLink) {
+	for _, step := range append(append(s.pre, s.tests...), s.post...) {
+		ret = append(ret, api.InternalImageLink(api.PipelineImageStreamTagReference(step.Image)))
+	}
+	return
+}
+
+func (s *newE2ETestStep) Creates() []api.StepLink { return nil }
+func (s *newE2ETestStep) Provides() (api.ParameterMap, api.StepLink) {
+	return nil, nil
+}
+func (s *newE2ETestStep) SubTests() []*junit.TestCase { return nil }
+
+func (s newE2ETestStep) runSteps(ctx context.Context, steps []api.TestStep) error {
+	pods, err := s.genPods(steps)
+	if err != nil {
+		return err
+	}
+	return s.runPods(ctx, pods)
+}
+
+func (s newE2ETestStep) genPods(steps []api.TestStep) ([]coreapi.Pod, error) {
+	var ret []coreapi.Pod
+	var errs []error
+	for _, step := range steps {
+		resources, err := resourcesFor(step.Resources)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+		name := s.name + "-" + step.Name
+		pod, err := generateBasePod(s.jobSpec, resources)
+		pod.ObjectMeta.Name = name
+		pod.ObjectMeta.Annotations[annotationContainersForSubTestResults] = step.Name
+		container := &pod.Spec.Containers[0]
+		container.Name = step.Name
+		container.Image = step.Image
+		container.Command = []string{"/bin/sh", "-c", "#!/bin/sh\nset -eu\n" + step.Commands}
+		container.Env = append(container.Env, []coreapi.EnvVar{
```

Comment:
I think this has changed (perhaps since you last looked) -- look for Hongkai's commit to see how to add all env vars

Reply:
Yes, the branch I linked above is based on https://github.com/openshift/release/commit/dc0e476cbb24354a30a58694687657b143aad698, which is a few weeks old now, but I looked briefly at the diff and didn't see anything major.

I really don't want to chase the templates, so I'll keep monitoring them for big changes but will only pull the changes when we are ready.